zerver/lib/addressee.py

```diff
@@ -105,6 +105,9 @@ def legacy_build(sender: UserProfile,
 
             return Addressee.for_stream(stream_name, topic_name)
         elif message_type_name == 'private':
+            if not message_to:
```

Comment:
I think this check should be in `for_emails`, so that you can strip and split `message_to` as needed.  You want to guard against `message_to` being something like "    " (which evaluates to `True` in python, I believe, but is still empty from our perspective).

Reply:
Okay, I'm not sure if I understand, why should this be in `recipient_for_emails`? This piece of code has nothing to do with that function (I think), right? Also, in Python, I think an empty string evaluates to `False`.

Reply:
Also, btw, all of the stripping is already done by `extract_recipients` anyway.

Reply:
I believe this commit is incorrect as written.  Not everything goes through `legacy_build`, and we don't have any explicit validation that the recipients list is non-empty in the `Addressee.for_private()` code path.  I think we just want to add an `assert len(message_to) > 0` sort of thing at the top of `Addressee.for_private` (I think it's OK to 500 in that case, because callers should be doing the appropriate validation), and do an audit of the callers to ensure they can never be empty.  By my read, there's exactly one that can be:

* `send_message` in `bot_lib.py` calls  `internal_send_huddle_message` with a potentially 0-length list.  I think we should fix that one inside `bot_lib.py`; it should be validating its input and returning an error in some appropriate fashion.
* (The other call to `internal_send_huddle_message` in production code is safe, because it starts with a valid Recipient object)

So we should fix that (changing `bot_lib.py` adding the `assert`), and then this commit is mergable.

Reply:
Just bumpinb this.